### PR TITLE
feat(pre-join): add new correlationId logic

### PR DESCRIPTION
**Pull request by torgeadelin**
# COMPLETES [SPARK-454228](https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-454228)

## This pull request addresses
- at the moment, correlationId is passed to the join method on the meeting object in SDK. This is too late in the flow. Updating correlationID in SDK should be done as part of the createMeeting method.
- needed for pre-join CA metrics 

## by making the following changes
- remove correlationId options from public join method to public create method.

### Change Type

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Tooling change
- [ ] Internal code refactor

## The following scenarios where tested
-unit test, manual test

### I certified that

- [x] I have read and followed [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request)
- [x] I discussed changes with code owners prior to submitting this pull request
- [x] I have not skipped any automated checks
- [x] All existing and new tests passed
- [ ] I have updated the documentation accordingly

---

Make sure to have followed the [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request) before submitting.

### CODE REVIEW

**Comment by torgeadelin** [comment]
Potentially breaking change (removing the options.correlationId) from public `join` method.
@arun3528

File: packages/@webex/plugin-meetings/src/meeting/index.ts [hunk 1]
```diff
@@ -593,7 +592,14 @@ export default class Meeting extends StatelessWebexPlugin {
      * @public
      * @memberof Meeting
      */
-    this.correlationId = this.id;
+    if (attrs.correlationId) {
+      LoggerProxy.logger.log(
+        `Meetings:index#constructor --> Initializing the meetin object with correlation id from app ${this.correlationId}`
```
Comment (jor-row) [inline packages/@webex/plugin-meetings/src/meeting/index.ts:597]: typo: ```meetin``` -> ```meeting```

**Review by jor-row — APPROVED** [review]

**Review by arun3528 — APPROVED** [review]

File: packages/@webex/plugin-meetings/src/meeting/index.ts [hunk 2]
```diff
@@ -4782,7 +4781,7 @@ export default class Meeting extends StatelessWebexPlugin {
       error instanceof Errors.SdpOfferHandlingError ||
       error instanceof Errors.SdpAnswerHandlingError
     ) {
-      sendBehavioralMetric(BEHAVIORAL_METRICS.PEERCONNECTION_FAILURE, error, this.id);
+      sendBehavioralMetric(BEHAVIORAL_METRICS.PEERCONNECTION_FAILURE, error, this.correlationId);
```
Comment (chrisadubois) [inline packages/@webex/plugin-meetings/src/meeting/index.ts:4784]: it seems so odd to me that we are sending behavioral metrics in the SDK with the work we are talking about with call analyzer... i know we added this long back... but wondering if it's still the move.

File: packages/@webex/plugin-meetings/src/meeting/index.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (torgeadelin) [reply packages/@webex/plugin-meetings/src/meeting/index.ts:597]: addressed

Comment (torgeadelin) [reply packages/@webex/plugin-meetings/src/meeting/index.ts]: always have the best eyes

File: packages/@webex/plugin-meetings/src/meeting/index.ts (hunk 2 above)
Comment (torgeadelin) [reply packages/@webex/plugin-meetings/src/meeting/index.ts:4784]: I know SDK team has a bunch of dashboards in Amplitude and wouldn't want to break them so will leave it here.

**Pull request merged**